fix(filetype): move fallback logic to `vim.filetype.match()` (#30141)
neovim/neovim: runtime/doc/lua.txt

        (`function?`) A function that modifies buffer state when called (for
        example, to set some filetype specific buffer variables). The function
        accepts a buffer number as its only argument.

>         (`boolean?`) True if a match was found via fallback

Reviewer: Also seems like there is no test coverage?

        (`function?`) A function that modifies buffer state when called (for
        example, to set some filetype specific buffer variables). The function
        accepts a buffer number as its only argument.

>         (`boolean?`) Return true if a match was found by falling back to a

Reviewer: Should it be a string in case there are other variants in the future?
```suggestion
        (`string?`) "FALLBACK" if a match was found by falling back to a
```

>         generic configuration file (".conf"). If true, the filetype should be
>         set with `:setf FALLBACK conf`, which enables a later `:setf` command
>         to override the filetype. See `:help setf` for more information.

Reviewer: is ".confg" an example?
```suggestion
        generic configuration file (e.g. ".conf"). If true, the filetype should
        be set with `:setf FALLBACK conf`, which enables a later |:setf| command
        to override the filetype.
```
